examples/gno.land/r/sys/users/store.gno

```diff
@@ -0,0 +1,186 @@
+package users
+
+import (
+	"regexp"
+	"std"
+	"strings"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/pausable"
+)
+
+var (
+	callerWhitelist = avl.NewTree() // address > struct{}{}
+
+	nameStore    = avl.NewTree() // name/aliases > *UserData
+	addressStore = avl.NewTree() // address > *UserData - always points to the latest alias
+
+	reAddressLookalike = regexp.MustCompile(`^g1[a-z0-9]{20,38}$`)
```

Comment:
I think it would be beneficial to expose this through a method like `IsValidateUsername`. Other libraries may want to pre-validate a username, which would help avoid multiple sources of truth regarding this regexp.